### PR TITLE
Merge dev to feat-2.0

### LINKED ISSUE #4113: merge dev to feat-2.0

### PR DESCRIPTION
**Pull request by real-felix**

### CODE REVIEW

File: execution_engine/src/engine_state/execution_result.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@ use casper_types::{
 };
 
 use super::error;
-use crate::{engine_state::ExecutionJournal, execution::Error as ExecError};
+use crate::{engine_state::execution_journal::ExecutionJournal, execution::Error as ExecError};
```
Comment (Fraser999) [inline execution_engine/src/engine_state/execution_result.rs]: I think this change should be reverted.

File: execution_engine/src/engine_state/genesis.rs [hunk 2]
```diff
@@ -32,7 +32,9 @@ use casper_types::{
 };
 
 use crate::{
-    engine_state::{execution_effect::ExecutionEffect, SystemContractRegistry},
+    engine_state::{
```
Comment (Fraser999) [inline execution_engine/src/engine_state/genesis.rs]: And this one.

File: execution_engine/src/engine_state/mod.rs [hunk 3]
```diff
@@ -7,7 +7,7 @@ pub mod era_validators;
 mod error;
 pub mod execute_request;
 pub mod execution_effect;
-mod execution_journal;
+pub(crate) mod execution_journal;
```
Comment (Fraser999) [inline execution_engine/src/engine_state/mod.rs]: And this one.

File: execution_engine/src/engine_state/step.rs [hunk 4]
```diff
@@ -7,7 +7,7 @@ use std::vec::Vec;
 use casper_types::{bytesrepr, CLValueError, Digest, EraId, ProtocolVersion, PublicKey};
 
 use crate::{
-    engine_state::{Error, ExecutionJournal},
+    engine_state::{error::Error, execution_journal::ExecutionJournal},
```
Comment (Fraser999) [inline execution_engine/src/engine_state/step.rs]: And this one.

File: execution_engine/src/engine_state/transfer.rs [hunk 5]
```diff
@@ -9,7 +9,7 @@ use casper_types::{
 };
 
 use crate::{
-    engine_state::Error,
+    engine_state::error::Error,
```
Comment (Fraser999) [inline execution_engine/src/engine_state/transfer.rs]: And this one.

File: execution_engine/src/execution/executor.rs [hunk 6]
```diff
@@ -12,7 +12,8 @@ use casper_types::{
 
 use crate::{
     engine_state::{
-        execution_kind::ExecutionKind, execution_result::ExecutionResult, EngineConfig, ExecError,
+        engine_config::EngineConfig, execution_kind::ExecutionKind,
+        execution_result::ExecutionResult,
```
Comment (Fraser999) [inline execution_engine/src/execution/executor.rs]: And this one (at least in part).

File: execution_engine/src/runtime/mod.rs [hunk 7]
```diff
@@ -44,7 +44,7 @@ use casper_types::{
 };
 
 use crate::{
-    engine_state::EngineConfig,
+    engine_state::engine_config::EngineConfig,
```
Comment (Fraser999) [inline execution_engine/src/runtime/mod.rs]: And this one.

File: execution_engine/src/runtime_context/mod.rs [hunk 8]
```diff
@@ -11,7 +11,8 @@ use tracing::error;
 
 use crate::{
     engine_state::{
-        execution_effect::ExecutionEffect, EngineConfig, ExecutionJournal, SystemContractRegistry,
+        engine_config::EngineConfig, execution_effect::ExecutionEffect,
+        execution_journal::ExecutionJournal, system_contract_registry::SystemContractRegistry,
```
Comment (Fraser999) [inline execution_engine/src/runtime_context/mod.rs]: And this one (at least in part).

File: execution_engine/src/runtime_context/tests.rs [hunk 9]
```diff
@@ -29,7 +29,7 @@ use tempfile::TempDir;
 
 use super::{Error, RuntimeContext};
 use crate::{
-    engine_state::{EngineConfig, SystemContractRegistry},
+    engine_state::{engine_config::EngineConfig, system_contract_registry::SystemContractRegistry},
```
Comment (Fraser999) [inline execution_engine/src/runtime_context/tests.rs]: And this one.

File: execution_engine/src/tracking_copy/ext.rs [hunk 10]
```diff
@@ -8,7 +8,9 @@ use casper_types::{
 };
 
 use crate::{
-    engine_state::{ChecksumRegistry, SystemContractRegistry},
+    engine_state::{
+        checksum_registry::ChecksumRegistry, system_contract_registry::SystemContractRegistry,
+    },
```
Comment (Fraser999) [inline execution_engine/src/tracking_copy/ext.rs]: And this one.

File: execution_engine/src/tracking_copy/mod.rs [hunk 11]
```diff
@@ -29,9 +29,11 @@ use casper_types::{
 
 pub use self::ext::TrackingCopyExt;
 use self::meter::{heap_meter::HeapSize, Meter};
-use super::engine_state::EngineConfig;
 use crate::{
-    engine_state::{execution_effect::ExecutionEffect, ExecutionJournal},
+    engine_state::{
+        engine_config::EngineConfig, execution_effect::ExecutionEffect,
+        execution_journal::ExecutionJournal,
+    },
```
Comment (Fraser999) [inline execution_engine/src/tracking_copy/mod.rs]: And this one (at least in part).

File: execution_engine/src/tracking_copy/tests.rs [hunk 12]
```diff
@@ -20,7 +20,7 @@ use super::{
     meter::count_meter::Count, AddResult, TrackingCopy, TrackingCopyCache, TrackingCopyQueryResult,
 };
 use crate::{
-    engine_state::{EngineConfig, ExecutionJournal},
+    engine_state::{engine_config::EngineConfig, execution_journal::ExecutionJournal},
```
Comment (Fraser999) [inline execution_engine/src/tracking_copy/tests.rs]: And this one.

**Review by Fraser999 — APPROVED** [review]
LGTM except for a bunch of nitpicks where we add back in unneeded EE qualifiers.

File: execution_engine/src/engine_state/execution_result.rs (hunk 1 above)
Comment (real-felix) [reply execution_engine/src/engine_state/execution_result.rs]: Done

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**